torch/_inductor/dependencies.py

```diff
@@ -49,7 +49,7 @@ def numbytes_hint(self):
             for var, size in zip(self.var_names, self.size):
                 if var in vars:
                     numel = numel * size
-        return V.graph.sizevars.size_hint(numel) * get_dtype_size(
+        return V.graph.sizevars.size_hint(numel, fallback=8192) * get_dtype_size(
```

Comment:
I don't think this is sound, see

```
    def can_inplace(self, read_dep: dependencies.MemoryDep):
        if self.get_aliases() or self.is_template():
            return False
                
        if read_dep.name not in self.scheduler.name_to_node:
            # don't allow reuse of an 'input' buffer, we don't own it
            # (would this have been fixed if I tracked mutations properly above?)
            return False
            
        if not isinstance(
            self.node, (torch._inductor.ir.AllReduce, torch._inductor.ir.InPlaceHint)
        ):
            # TODO make this a property of the IR
            return False

        if len(self.read_writes.writes) == 1:
            write_dep = next(iter(self.read_writes.writes))
            return read_dep.numbytes_hint() == write_dep.numbytes_hint()

        return False
```

We must know for sure that the sizes are the same to inplace, so we shouldn't fallback, we should reject inplacing (or maybe do it symbolically?)

Reply:
Is the problem you're referring to specifically related to `can_inplace` (in which case we can, indeed, reject inplacing in `can_inplace` in the face of unbacked SymInts)? Or are there other cases where using the `fallback=8192` inside `numbytes_hint` can cause problems (in which case we can make `fallback=None` a parameter of `numbytes_hint`, propagate it to `size_hint`, and set it to `8192` only from unproblematic places)?

Reply:
`can_inplace` uses `numbytes_hint` which is where you made the fallback use. So you  just need to fix `can_inplace` to use something else. But it looks like this function is actually the sole user.